src/NetEscapades.AspNetCore.SecurityHeaders/SecurityHeadersMiddleware.cs

```diff
@@ -64,16 +64,14 @@ public async Task Invoke(HttpContext context)
             context.SetNonce(_nonceGenerator.GetNonce(Constants.DefaultBytesInNonce));
         }
 
-        context.Response.OnStarting(OnResponseStarting, Tuple.Create(this, context, _policy));
+        context.Response.OnStarting(() => OnResponseStarting(context));
```

Comment:
This avoids the `Tuple` allocation (it's not boxing, right, because it's a `Tuple` not a `ValueTuple`?), but it replaces it with a capture of the `context` and the lambda, so it still allocates almost as much I think? It seems like this would have a very minimal benefit, if any?

Reply:
Yes, creating a Tuple is not boxing. But you are passing the Tuple to the OnResponseStarting method which expects an object called state. So when you pass the Tuple to this method, it is boxed to object. And inside the method, it is unboxed back to Tuple.
As for allocation, yes, almost the same memory because Tuple uses ref. The main point here is that you are passing a class object to a class method, which is totally fine, but you don't need it.

Reply:
Ok sure, when people talk about boxing they're normally talking about normally talking about the allocation that comes from boxing a struct. 

In general I'm hesitant to make outstanding performance-related changes without Ann infusion it will improve performance 🙂 but I'll have a think, thanks

Reply:
Makes sense. In this matter, I think I can help.

I created a benchmark, using BenchmarkDotNet, with both codes and ran the same code of the HttpRequest_WithDefaultSecurityHeaders_SetsSecurityHeaders test.

- **SecurityHeadersWithTupleAndStaticMethod** represents using Tuple and executing static method.
- **SecurityHeadersWithObjPropsAndNonStaticMethod** represents using object properties, forwarding context and executing non-static method.

This is the result I got:

![image](https://github.com/user-attachments/assets/2e0d7f5c-3087-4114-bf46-af359d4951fd)

As I mentioned, the benefits are minimal.

I haven't committed the benchmark code to the repository to avoid clutter and pollution, but I can share it with you if you find it interesting.